File: A4/src/stack/MyStack.java
package stack;
import java.util.Stack;

public class MyStack <T> implements MyIStack <T>
{
	Stack<T> stack;
	
	public MyStack ()
	{
		this.stack = new Stack<T>();
	}
	@Override
	public T pop()
	{
		return stack.pop();
	}

	@Override
	public void push(T v)
	{
		stack.push(v);
	}

	@Override
	public boolean isEmpty()
	{
		return stack.isEmpty();
	}
	
	public String toString()
	{
		String result = "\n";
		for (int i = stack.size() - 1; i >= 0; i--)
			result += stack.get(i).toString() + "\n";
		return result;
	}
}

File: A7/Stack/MyStack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace A7.Stack
{
    class MyStack<T>: MyIStack<T>
    {
        Stack<T> stack;
        public MyStack()
        {
            this.stack = new Stack<T>();
        }
        public T Pop()
        {
            return stack.Pop();
        }
        public void Push(T e)
        {
            stack.Push(e);
        }

        public Boolean IsEmpty()
        {
            return stack.Count() == 0;
        }
        public override String ToString()
        {
            String result = "\n";
            foreach (T el in stack)
            {
                result += el.ToString() + Environment.NewLine;
            }
            return result;
        }
    }
}
